Neos.ContentRepository.Core/Classes/Feature/WorkspaceCommandHandler.php

```diff
@@ -317,10 +317,7 @@ private function publishContentStream(
                 $copiedEvent = $event->createCopyForContentStream($baseContentStreamId);
                 // We need to add the event metadata here for rebasing in nested workspace situations
                 // (and for exporting)
-                $events[] = DecoratedEvent::withMetadata(
-                    $copiedEvent,
-                    $eventEnvelope->event->metadata
-                );
+                $events[] = $eventEnvelope->event->metadata !== null ? DecoratedEvent::withMetadata($copiedEvent, $eventEnvelope->event->metadata) : $copiedEvent;
```

Comment:
This seems to be a sudden change: `$events` contains `DecoratedEvent` AND `EventInterface` i presume now. Is this correct?

Reply:
That's right, but it doesn't matter.
But, good point, we currently ignore potential causation/correlation ids from the copied event.. I'll check

Reply:
Haha thanks for checking ^^